### ISSUE TITLE
FASTLANE_USERNAME and FASTLANE_PASSWORD are missleading

### ISSUE THREAD

**Issue by jlo-photobox** [fastlane/fastlane#10382]
### New Issue Checklist

- [*] Updated fastlane to the latest version
- [*] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [*] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [*] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

The FASTLANE_USERNAME and FASTLANE_PASSWORD environment variable names are missleading, since it doesn't clearly convey that it is used as an iTunes connect credential.

My problem with it has manifested when someone changed the password of the iTunes connect account we share, and CI started failing.

MATCH_PASSWORD was the first guess on where that password should live, however I discovered that it turns out to be the passphrase to decrypt the profiles and certificates. This is documented in Match's README but the name could also be reconsidered.

Having given up on this and not found a solution in the README, I verified via trial and error that Match creates in the keychain the deliver.... password, I updated that and the build passed locally, but not on CI.

It was only after a while of trial and error that I ended up realising that FASTLANE_PASSWORD was the field I should update, and that fixed our CI.

In summary, I wasted a bunch of hours due to the naming of environment variables being missleading, and IMO it should be revamped to be self explanatory.

### Environment

Please run `fastlane env` and copy the output below. This will help us help you :+1:
If you used `--capture_output` option please remove this block - as it is already included there.

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.12.6                                     |
| Ruby                        | 2.3.0                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.13.1                          |
| Installation Source         | ~/.rvm/gems/ruby-2.3.0/bin/fastlane         |
| Host                        | Mac OS X 10.12.6 (16G29)                    |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.3.0/lib                |
| OpenSSL Version             | OpenSSL 1.0.2l  25 May 2017                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 8.3.3                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_GB.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status      |
| -------- | ------- | ------------------ |
| fastlane | 2.53.1  | 🚫 Update available |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.0.0        |
| executable-hooks          | 1.3.2        |
| bundler-unload            | 1.0.2        |
| rubygems-bundler          | 1.4.4        |
| bundler                   | 1.15.1       |
| io-console                | 0.4.5        |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.5        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.5        |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| declarative               | 0.0.9        |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.1        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| httpclient                | 2.8.3        |
| google-api-client         | 0.12.0       |
| mini_magick               | 4.5.1        |
| multi_json                | 1.12.1       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| json                      | 2.1.0        |
| excon                     | 0.58.0       |
| plist                     | 3.3.0        |
| faraday                   | 0.12.2       |
| jwt                       | 1.5.6        |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.3        |
| faraday_middleware        | 0.12.2       |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.1        |
| terminal-table            | 1.8.0        |
| unicode-display_width     | 1.3.0        |
</details>

*generated on:* **2017-09-20**
</details>

**Comment by KrauseFx** [comment]
I'm really sorry you had those problems - I completely understand your concerns. It's a little hard to change environment variables once they're being used, as this would break existing setups. So it seems like the best approach to solve this would be to improve the docs, right?

**Comment by jlo-photobox** [comment]
No need to be sorry 😛 

Yeah I agree on improving docs only. I was thinking if it'd be possible to deprecate it and support both old env. vars and new ones, but I guess that might be a change far too complex for such a small benefit.

Cheers on a great tool btw 👍
